pysmt/rewritings.py

```diff
@@ -1223,6 +1224,194 @@ def __init__(self, environment=None):
+        result = self.mgr.Div(result,base)
+        return result
+
+    def process_real_round(self,realval,om,rm,sign,total_width,frac_width):
+        tempres = self.round_real(realval,rm,frac_width)
+        result = self.process_real_noround(tempres,om,sign,total_width,frac_width)
+        return result
+
+
+
+    def walk_ufxp_add(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        left = args[1]
+        right = args[2]
+        result = self.mgr.Plus(left,right)
+        return self.process_real_noround(result,om,0,total_width,frac_width)
+
+    
+
+    def walk_sfxp_add(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        left = args[1]
+        right = args[2]
+        result = self.mgr.Plus(left,right)
+        return self.process_real_noround(result,om,1,total_width,frac_width)
+        
+
+
+    def walk_ufxp_sub(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        left = args[1]
+        right = args[2]
+        result = self.mgr.Minus(left,right)
+        return self.process_real_noround(result,om,0,total_width,frac_width)
+        
+
+    def walk_sfxp_sub(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        left = args[1]
+        right = args[2]
+        result = self.mgr.Minus(left,right)
+        return self.process_real_noround(result,om,1,total_width,frac_width)
+
+    def walk_ufxp_mul(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        rm = args[1]
+        left = args[2]
+        right = args[3]
+        result = self.mgr.Times(left,right)
+        return self.process_real_round(result,om,rm,0,total_width,frac_width)
+
+    def walk_sfxp_mul(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        rm = args[1]
+        left = args[2]
+        right = args[3]
+        result = self.mgr.Times(left,right)
+        return self.process_real_round(result,om,rm,1,total_width,frac_width)
+
+    def walk_ufxp_div(self, formula, args, **kwargs):
+        ty = self.env.stc.get_type(formula)
+        total_width = ty.total_width
+        frac_width = ty.frac_width
+        om = args[0]
+        rm = args[1]
+        left = args[2]
+        right = args[3]
+        result = self.mgr.Div(left,right)
+        return self.process_real_round(result,om,rm,0,total_width,frac_width)
```

Comment:
This needs to handle the case of a zero divisor:
self.mgr.Ite(self.mgr.Equals(divisor, zero), Fraction(2**total_width-1), ...)
Likewise for sfxp_div

Reply:
Actually we need to change PySMT to make this work.